### ISSUE TITLE
bug(YouTube Music): app crashes on Android 5, 6

### ISSUE THREAD

**Issue by jpa102** [inotia00/ReVanced_Extended#2449]
### Application

YouTube Music

### Question description

since i saw in the description that the 6.20.51 version of YT Music can be patched, i tried patching the apk for fun with your patches and install it on my secondary phone

if i patch the 6.20.51 version of YT Music (which is in the minimum version) on my android 13 phone, it installs and runs just fine

now if i take the same patched apk and install it on my android 5 / 5.1 phone, it installs but it crashes on startup

why does it crash on my other phone? given that the 6.20.51 apk can support android 5 (source: apkmirror) which is the minimum android version, it should run without any problems. i even tried patching the v8a and v7a apk variants without success

i have no idea if this is only happening on my secondary phone

for some context, my main phone is samsung galaxy m12 (android 13) and my other phone is oppo a37 (android 5). both phones have revanced gmscore installed and they are not rooted

### Acknowledgements

- [X] This issue is not a duplicate of an existing question.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by inotia00** [comment]
Unfortunately I don't have an Android 5 device to reproduce the issue

Can you attach the crash log?

**Comment by UnknownErrorNotFound** [comment]
You need to use your other phone to patch YouTube, if you patched YouTube with RVX Manager it removes unnecessary arhcs, which means that it can not run on your other phone because arch is missing from the patched apk.

**Comment by jpa102** [comment]
> Can you attach the crash log?

sure, i'll try to get a crash log via adb

**Comment by jpa102** [comment]
> Or you can just switch off this setting in RVX Manager, this way no archs are not removed

didn't realize that setting, i'll try repatching YT Music

if this still happens, i'll reply back with a crash log

**Comment by jpa102** [comment]
the crashing still happens, but i'm not able to get logs from adb unfortunately because i don't have any other working charger cable at the moment to connect to my desktop

i have tried turning off the "Remove unused library" setting, but that didn't made a difference

**Comment by jpa102** [comment]
alright, i now have the crashlogs

i tried opening up the app twice though

```
shell@A37f:/ $ logcat | grep AndroidRuntime
D/AndroidRuntime(23526): Shutting down VM
E/AndroidRuntime(23526): FATAL EXCEPTION: main
E/AndroidRuntime(23526): Process: app.rvxvanced.android.apps.youtube.music, PID: 23526
E/AndroidRuntime(23526): java.lang.NoSuchMethodError: No interface method computeIfAbsent(Ljava/lang/Object;Ljava/util/fears in /system/framework/core-libart.jar)
E/AndroidRuntime(23526):        at app.revanced.integrations.shared.utils.StringRef.sfc(StringRef.java:29)
E/AndroidRuntime(23526):        at app.revanced.integrations.shared.utils.StringRef.str(StringRef.java:52)
E/AndroidRuntime(23526):        at app.revanced.integrations.music.utils.RestartUtils.showRestartDialog(RestartUtils.jav
E/AndroidRuntime(23526):        at app.revanced.integrations.music.patches.utils.InitializationPatch.onCreate(Initializa
E/AndroidRuntime(23526):        at com.google.android.apps.youtube.music.activities.MusicActivity.onCreate(PG)
E/AndroidRuntime(23526):        at android.app.Activity.performCreate(Activity.java:6093)
E/AndroidRuntime(23526):        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
E/AndroidRuntime(23526):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2458)
E/AndroidRuntime(23526):        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2577)
E/AndroidRuntime(23526):        at android.app.ActivityThread.access$1000(ActivityThread.java:164)
E/AndroidRuntime(23526):        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1462)
E/AndroidRuntime(23526):        at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(23526):        at android.os.Looper.loop(Looper.java:160)
E/AndroidRuntime(23526):        at android.app.ActivityThread.main(ActivityThread.java:5541)
E/AndroidRuntime(23526):        at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(23526):        at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(23526):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:964)
E/AndroidRuntime(23526):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:759)
D/AndroidRuntime(23668): Shutting down VM
E/AndroidRuntime(23668): FATAL EXCEPTION: main
E/AndroidRuntime(23668): Process: app.rvxvanced.android.apps.youtube.music, PID: 23668
E/AndroidRuntime(23668): java.lang.NoSuchMethodError: No interface method computeIfAbsent(Ljava/lang/Object;Ljava/util/fears in /system/framework/core-libart.jar)
E/AndroidRuntime(23668):        at app.revanced.integrations.shared.utils.StringRef.sfc(StringRef.java:29)
E/AndroidRuntime(23668):        at app.revanced.integrations.shared.utils.StringRef.str(StringRef.java:52)
E/AndroidRuntime(23668):        at app.revanced.integrations.music.utils.RestartUtils.showRestartDialog(RestartUtils.jav
E/AndroidRuntime(23668):        at app.revanced.integrations.music.patches.utils.InitializationPatch.onCreate(Initializa
E/AndroidRuntime(23668):        at com.google.android.apps.youtube.music.activities.MusicActivity.onCreate(PG)
E/AndroidRuntime(23668):        at android.app.Activity.performCreate(Activity.java:6093)
E/AndroidRuntime(23668):        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
E/AndroidRuntime(23668):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2458)
E/AndroidRuntime(23668):        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2577)
E/AndroidRuntime(23668):        at android.app.ActivityThread.access$1000(ActivityThread.java:164)
E/AndroidRuntime(23668):        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1462)
E/AndroidRuntime(23668):        at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(23668):        at android.os.Looper.loop(Looper.java:160)
E/AndroidRuntime(23668):        at android.app.ActivityThread.main(ActivityThread.java:5541)
E/AndroidRuntime(23668):        at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(23668):        at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(23668):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:964)
E/AndroidRuntime(23668):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:759)
```

**Comment by jpa102** [comment]
successfully patched using `dev-release` on rvx manager `1.22.2`

strange that it fails in applying my own custom logo

other than the first crash at startup, it now works

![Screenshot_2024-10-27-19-56-17-54](https://github.com/user-attachments/assets/3e15a214-f5a4-44e0-886d-87d81db395cb)

**Comment by inotia00** [comment]
reflected in [revanced-patches-v4.16.1](https://github.com/inotia00/revanced-patches/releases/tag/v4.16.1)
